cmd/tunneld/tunneld.go

```diff
@@ -138,3 +150,87 @@ func fatal(format string, a ...interface{}) {
 	fmt.Fprint(os.Stderr, "\n")
 	os.Exit(1)
 }
+
+func startAutocert(opts *options, server *tunnel.Server, logger log.Logger) {
+	cacheDir, err := cacheDir(opts.letsEncryptCacheDir)
+	if err != nil {
+		fatal("failed to set cache directory for certs ", err)
+	}
+	m := &autocert.Manager{
+		Prompt:     autocert.AcceptTOS,
+		HostPolicy: hostPolicy(server),
+		Cache:      autocert.DirCache(cacheDir),
+	}
+	if opts.httpAddr == ":80" {
```

Comment:
All the validation shall be taken out from here and move do `options.go`. Please add a `validate` method to `options` and check it before any processing.